src/devices/Ak8963/Ak8963.cs

```diff
@@ -0,0 +1,287 @@
+
+        /// <summary>
+        /// Default constructor for an independent AK8963
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device</param>
+        public Ak8963(I2cDevice i2CDevice) : this(i2CDevice, new Ak8963I2c())
+        { }
+
+        /// <summary>
+        /// Constructor to use if AK8963 is behind another element and need a special I2C protocol like
+        /// when used with the MPU9250
+        /// </summary>
+        /// <param name="i2CDevice">The I2C device</param>
+        /// <param name="ak8963Interface">The specific interface to communicate with the AK8963</param>
+        /// <param name="autoDispose">True to dispose the I2C device when class is disposed</param>
+        public Ak8963(I2cDevice i2CDevice, Ak8963I2cBase ak8963Interface, bool autoDispose = true)
+        {
+            _i2cDevice = i2CDevice;
+            _ak8963Interface = ak8963Interface;
+            // Initialize the default modes
+            _measurementMode = MeasurementMode.PowerDown;
+            _outputBitMode = OutputBitMode.Output14bit;
+            _autodispose = autoDispose;
+            byte mode = (byte)((byte)_measurementMode | ((byte)_outputBitMode << 4));
+            WriteRegister(Register.CNTL, mode);
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x48 for a AK8963");
+        }
+
+        /// <summary>
+        /// Reset the device
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.RSV, 0x01);
+            // When powering the AK893, doc says 50 ms needed
+            Thread.Sleep(50);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns>The device information</returns>
+        public byte GetDeviceInfo()
+        {
+            return ReadByte(Register.INFO);
+        }
+
+        /// <summary>
+        /// Get the magnetometer bias
+        /// </summary>
+        public Vector3 MagnometerBias { get; set; } = Vector3.Zero;
+
+        /// <summary>
+        /// Calibrate the magnetometer. Make sure your sensor is as far as possible of magnet
+        /// Calculate as well the magnetometer bias
+        /// </summary>
+        /// <returns>Returns the factory calibration data</returns>
+        public Vector3 CalibrateMagnetometer()
+        {
+            Vector3 calib = new Vector3();
+            Span<byte> rawData = stackalloc byte[3];
+
+            var oldPower = MeasurementMode;
+
+            // Stop the magnetometer
+            MeasurementMode = MeasurementMode.PowerDown;
+            // Enter the magnetometer Fuse mode to read the calibration data
+            // Page 13 of documentation 
+            MeasurementMode = MeasurementMode.FuseRomAccess;
+            // Read the data
+            ReadBytes(Register.ASAX, rawData);
+            calib.X = (float)((rawData[0] - 128) / 256.0 + 1.0);
```

Comment:
I find this calculation a bit confusing:
First you move `0..255` range to `-128..127` range, then you scale the range to around `-0.5..0.5` then you move it to 0.5..1.5? My gut feeling tells me that `+1` is incorrect or it should be `+0.5`

Reply:
hum, I need to dig into that and check that into details. I made quite a lot of changes in this. So that may not have been correct. Documentation on calibration is weak. 
Thanks for spotting it.